File: Data Structures & Algorithms/reverse-integer/submission-2.js
class Solution {
    /**
     * @param {number} x
     * @return {number}
     */
    reverse(x) {
        let ans = 0;
        let neg = x< 0;
        let max = Math.pow(2,31)-1
        if(x<0) x= -x;
        while(x>0){
            ans  = ans*10 + x%10 ;
            x =Math.floor( x/10);
            if(ans > max ) return 0;
        }
        return(neg  ? -ans : ans );
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum  = 0  ;
        int ans = 0 ;
        int max  =  nums[0];
        for(int x : nums){
            sum+=x ;
            if(sum < 0) sum = 0 ;
            ans = Math.max(ans,sum);
            max = Math.max(max,x);
        }
        return ans==0 ? max : ans  ;
    }
}
